internal/controller/driver_controller.go

```diff
@@ -1259,3 +1299,55 @@ func mergeDriverSpecs(dest, src *csiv1a1.DriverSpec) {
 		dest.CephFsClientType = src.CephFsClientType
 	}
 }
+
+func (r *driverReconcile) logrRotateContainer(imagePullPolicy corev1.PullPolicy, csiComponentName string, resources *corev1.ResourceRequirements) corev1.Container {
+	rotation := r.driver.Spec.Log.Rotation
+	return corev1.Container{
+		Name:            "log-collector",
+		Image:           r.images["plugin"],
+		ImagePullPolicy: imagePullPolicy,
+		Resources: ptr.Deref(
+			resources,
+			corev1.ResourceRequirements{},
+		),
+		Command: []string{
+			"/bin/bash",
+			"-c", // Command to run
+			fmt.Sprintf(cronLogRotate, rotation.Periodicity, rotation.MaxLogSize.String(), strconv.Itoa(rotation.MaxFiles), rotation.LogHostPath, csiComponentName),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			utils.LogDirVolumeMount(r.driver.Spec.Log.Rotation.LogHostPath, csiComponentName),
+			utils.LogSidecarVolumeMount(r.driver.Spec.Log.Rotation.LogHostPath, csiComponentName),
+		},
+	}
+}
+
+var cronLogRotate = `
+echo "Starting the csi-logrotate-sidecar"
+PERIODICITY=%s
+LOG_MAX_SIZE=%s
+ROTATE=%s
+CsiLogHostPath=%s
+CsiComponentName=%s
+
+mkdir -p "$CsiLogHostPath"/logrotate-config/"$CsiComponentName"
+echo '"$CsiLogHostPath"/log/"$CsiComponentName"/*.log {
+	"$PERIODICITY"
+	missingok
+	rotate "$ROTATE"
+	compress
+	copytruncate
+	notifempty
+}' >  "$CsiLogHostPath"/logrotate-config/"$CsiComponentName"/csi
+echo "File creation container completed"
+
+LOG_ROTATE_CEPH_CSI_FILE="$CsiLogHostPath"/logrotate-config/"$CsiComponentName"/csi
+if [ "$LOG_MAX_SIZE" != "0" ]; then
+  sed --in-place "4i \ \ \ \ maxsize $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_CSI_FILE"
+fi
+
+while true; do
+  logrotate --verbose "$LOG_ROTATE_CEPH_CSI_FILE"
+  sleep 15m
+done
+`
```

Comment:
Why do we have an inline bash script in our go code? 
If you need to have a log rotate sidecar build it as an image so it can be controlled by all of the image control mechanisms we already have

Reply:
Rook has had very good success using inline bash scripts to help tie functionality together. 

For one, it allows using any desired image, which helps with avoiding unnecessary pod restarts. For example, using the ceph-csi-operator container here means that the deployment/daemonset needs to be restarted with every new operator version. In contrast, if the ceph-csi image is used here, the restart is only needed when that image is already changing.

For a second, having dedicated images for stuff like this ends up being cumbersome and limits developer flexibility. When simple scripts are used, code maintenance goes a lot more quickly than in the past where Rook was using golang daemons for everything. It also means that any potential script bugs can be fixed in the operator and don't need to be fixed on a per-CSI-release basis.

If you are proposing to add this functionality to the ceph-csi image natively, we can probably work together to help implement that, but I still think the inline script provides good flexibility and transparency, and I think that is a consideration worth making.

But if you are proposing to create an entirely new image, I would push back to say that I don't think Parth should be the one managing that complexity.

Reply:
@nb-ohad this has been a topic of discussion in the Rook huddle for a couple days. I want to make sure my prior comment didn't get lost in the amount of other comments here. What are your and @Madhu-1 's thoughts on this?

Reply:
@BlaineEXE Regardless of the reasons, which we had several discussions on offline and which are mainly security concerns, we decided to shift to having the log-rotate config map be created by the operator and mounted into the pod using a volume. That by itself, make any kind of injected scripting unnecessary for this specific use case